exercise/0-IH-exercise/main.go

```diff
@@ -1,9 +1,54 @@
 package main
 
+import (
+	"fmt"
+	"runtime"
+	"time"
+)
+
+
 func main() {
-	// 3 task concurrent
-	// pertask beda2 sleepnya
-	// gunakan wait group dan channel
-	// waitnya tidak menunggu dari 5 detik
-	// cancelation bisa menggunakn time, context, channel
+	// use channel
+
+	runtime.GOMAXPROCS(2)
+
+	timeStart := time.Now()
+
+	var messages = make(chan string)
+
+	go taskOne(3, messages)
+	go taskTwo(2, messages)
+	go taskThree(5, messages)
+
+	var task1 = <-messages
+	fmt.Println(task1)
+
+	var task2 = <-messages
+	fmt.Println(task2)
+
+	var task3 = <-messages
+	fmt.Println(task3)
+
+	fmt.Println(time.Since(timeStart))
+}
+
+func taskOne(till int, messages chan string) {
```

Comment:
minor nitpick :) 
I usually prefer function, if they need time parameter, to accept `time.Duration`
```
func taskOne(duration time.Duration, messages chan string) {
```
so the caller has greater control on the timing loginc `go taskOne(time.Milliseconds * 100, messages)`